extensions/wikia/Flags/controllers/FlagsApiController.class.php

```diff
@@ -0,0 +1,336 @@
+<?php
+
+/**
+ * The public interface of the extension and the main entry point for all requests.
+ * It provides a set of CRUD methods to manipulate Flags instances and their types.
+ *
+ * @author Adam Karmiński <adamk@wikia-inc.com>
+ * @copyright (c) 2015 Wikia, Inc.
+ * @license http://www.gnu.org/copyleft/gpl.html GNU General Public License 2.0 or later
+ */
+
+use Flags\FlagsCache;
+use Flags\Models\Flag;
+use Flags\Models\FlagType;
+
+class FlagsApiController extends WikiaApiController {
+	private
+		$cache,
+		$params,
+		$status = false;
+
+	/**
+	 * Returns a singleton instance of FlagsCache
+	 * @return FlagsCache
+	 */
+	private function getCache() {
```

Comment:
This is just a suggestion, since we don't have a convention for it. But I generally think it's preferable to order methods by visibility (public methods all first, then protected and private), which is a common convention. It makes it easy to separate the API from the implementation. Right now the public-facing API is split by various private methods.